### PR TITLE
ARROW-9062: [Rust] json reader dictionary support

### PR DESCRIPTION
**Pull request by svenwb**
Changes in this PR allow the JSON Reader to read into Arrow arrays of type dictionary.

### CODE REVIEW

**Review by vertexclique — COMMENTED** [review]
Looks nice, ornamental changes.

File: rust/arrow/src/json/reader.rs [hunk 1]
```diff
@@ -448,42 +453,55 @@ impl<R: Read> Reader<R> {
...
-                                                    None
-                                                } else {
-                                                    Some(v.to_string())
-                                                }
-                                            }).collect()
-                                        } else if let Value::Null = value {
-                                            vec![None]
-                                        } else if !value.is_object() {
-                                            vec![Some(value.to_string())]
+                                    // value can be an array or a scalar
```
Comment (svenwb) [inline rust/arrow/src/json/reader.rs:456]: This was only changed by the formatter

File: rust/arrow/src/json/reader.rs [hunk 2]
```diff
@@ -448,42 +453,55 @@ impl<R: Read> Reader<R> {
...
+                                                Some(v.to_string())
+                                            }
+                                        }).collect()
+                                    } else if let Value::Null = value {
+                                        vec![None]
+                                    } else if !value.is_object() {
+                                        vec![Some(value.to_string())]
+                                    } else {
+                                        return Err(ArrowError::JsonError("Only scalars are currently supported in JSON arrays".to_string()));
+                                    };
```
[the remark this answers is not in the document]
Comment (svenwb) [reply rust/arrow/src/json/reader.rs:477]: As this code was only changed by the formatter I would suggest to make this refactoring in different PR.

**Review by nevi-me — APPROVED** [review]
LGTM, thanks @svenwb. I'll wait for @vertexclique to have another look at your changes first

**Review by vertexclique — APPROVED** [review]
Ok, looks good.

**Comment by svenwb** [comment]
Hey @nevi-me as @vertexclique approved as well, can you merge this? Or how shall we proceed? Thanks, Sven

**Pull request closed without merging**